x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_grid_page.tsx

```diff
@@ -5,758 +5,868 @@
-
-    const description = this.determineDescription(isAdmin, this.props.readOnly ?? false);
-
-    return (
-      <>
-        <KibanaPageTemplate.Header
-          paddingSize="none"
-          bottomBorder
-          pageTitle={
-            <FormattedMessage
-              id="xpack.security.management.apiKeys.table.apiKeysTitle"
-              defaultMessage="API Keys"
-            />
-          }
-          description={description}
-          rightSideItems={
-            this.props.readOnly
-              ? undefined
-              : [
-                  <EuiButton
-                    {...reactRouterNavigate(this.props.history, '/create')}
-                    fill
-                    iconType="plusInCircleFilled"
-                    data-test-subj="apiKeysCreateTableButton"
-                  >
-                    <FormattedMessage
-                      id="xpack.security.management.apiKeys.table.createButton"
-                      defaultMessage="Create API key"
-                    />
-                  </EuiButton>,
-                ]
-          }
-        />
+            }
+            rightSideItems={
+              !readOnly
+                ? [
+                    <EuiButton
+                      {...reactRouterNavigate(history, '/create')}
+                      fill
+                      iconType="plusInCircleFilled"
+                      data-test-subj="apiKeysCreateTableButton"
+                    >
+                      <FormattedMessage
+                        id="xpack.security.management.apiKeys.table.createButton"
+                        defaultMessage="Create API key"
+                      />
+                    </EuiButton>,
+                  ]
+                : undefined
+            }
+            paddingSize="none"
+            bottomBorder
+          />
+          <EuiSpacer />
+          <KibanaPageTemplate.Section paddingSize="none">
+            {createdApiKey && !state.loading && (
+              <>
+                <ApiKeyCreatedCallout createdApiKey={createdApiKey} />
+                <EuiSpacer />
+              </>
+            )}
 
-        {this.state.createdApiKey && !this.state.isLoadingTable && (
-          <>
-            <EuiSpacer size="l" />
-            <EuiCallOut
-              color="success"
-              iconType="check"
-              title={i18n.translate('xpack.security.management.apiKeys.createSuccessMessage', {
-                defaultMessage: "Created API key '{name}'",
-                values: { name: this.state.createdApiKey.name },
-              })}
+            {!state.value.canManageApiKeys ? (
+              <>
+                <EuiCallOut
+                  title={
+                    readOnly ? (
+                      <FormattedMessage
+                        id="xpack.security.management.apiKeys.table.readOnlyOwnKeysWarning"
+                        defaultMessage="You only have permission to view your own API keys."
```

Comment:
This message should not appear in read-only mode. The `read_security` privilege grants read access to all API keys, and the read-only UI denoted with the glasses icon (on top of having no ability to create/update/delete).

Reply:
Good point, I've fixed the condition to not show this when viewed in readonly mode.